src/main/java/net/neoforged/neoforge/registries/DeferredBlock.java

```diff
@@ -18,6 +19,25 @@
  * @param <T> The specific {@link Block} type.
  */
 public class DeferredBlock<T extends Block> extends DeferredHolder<Block, T> implements ItemLike {
+    /**
+     * Creates a new {@link ItemStack} with a default size of 1 from this {@link Block}
+     */
+    public ItemStack toStack() {
+        return toStack(1);
+    }
+
+    /**
+     * Creates a new {@link ItemStack} with the given size from this {@link Block}
+     *
+     * @param count The size of the stack to create
+     */
+    public ItemStack toStack(int count) {
+        ItemStack stack = asItem().getDefaultInstance();
+        if (stack.isEmpty()) throw new IllegalStateException("Item stack can't be null!");
```

Comment:
The error message is misleading, since the item stack isn't null (item stacks normally are never null), but that the block has no corresponding item stack, in which case `asItem()` returns `Items.AIR`. A better message would be something akin to "Block does not have a corresponding item" (without the exclamation point).

Reply:
done!